Userland/Libraries/LibWeb/WebAudio/DynamicsCompressorNode.h

```diff
@@ -17,6 +17,11 @@ struct DynamicsCompressorOptions : AudioNodeOptions {
     float ratio { 12 };
     float release { 0.25 };
     float threshold { -24 };
+    // DynamicsCompressor cannot have Max value (default for generic AudioNode)
+    DynamicsCompressorOptions()
+    {
+        channel_count_mode = Bindings::ChannelCountMode::ClampedMax; // Set ClampedMax only for DynamicsCompressorNode
+    }
```

Comment:
The definition for this struct comes straight from the spec - you should implement its behaviors in the code using explicit spec steps instead.

Reply:
> The definition for this struct comes straight from the spec - you should implement its behaviors in the code using explicit spec steps instead.

Yeah, that what I wanted to do. However, I don't quite understand how to inherit Audio Node Options without corresponding default values? The default value that is set at the moment is invalid according to the spec

Reply:
I'd make the default value part of the constructor, by calling the setter in construct_impl after creating the instance. Not sure if we have a good example anywhere else on how to do this.

Reply:
In constructor, how I distinguish between the situation constructor (in js) sets Max (we must fail) and default value (Max) is passed from audio node (we must change that to ClampedMax)?

Reply:
I cannot manage to move default value to constructor. The problem is, DynamicsCompressorOptions is passed to the constructor as is, no matter if it's default value (as per audio node spec) or user-defined value (I suppose set somewhere in parser)
If it's default value, as I understand, you suggest changing it to valid default value (as per spec), that is ClampedMax. However, I do not see a way to pass both "user set, fail" and "default value, not fail" scenarios from within the constructor